File: 알고리즘/권하경/20-05-29/20-05-29.js
function solution(skill, skill_trees) {
  let answer=0;
  skill_trees.forEach(element => {
    let index=0;
    for(let i in element){
      if(element[i]==skill[index])  index++;
      else if(element[i]!=skill[index]&&skill.includes(element[i]))  break;
      if(i==element.length-1||index==skill.length-1) {
        answer++;
        break;
      }
    }
  });
  return answer;
}

console.log(solution(	"CBD",["BACDE", "CBADF", "AECB", "BDA"]));

File: 알고리즘/권하경/20-05-29/20-05-29.java
class Solution {
    public static int solution(String skill, String[] skill_trees) {
        int answer = 0;
        for (String element:skill_trees) {
            int index =0;
            for (int i=0;i<element.length();i++){
                if(element.charAt(i)==skill.charAt(index)) index++;
                else if(element.charAt(i)!=skill.charAt(index)&&skill.indexOf(element.charAt(i))!=-1) break;
                if(i==element.length()-1||index==skill.length()) {
                    answer++;
                    break;
                }
            }
        }
        return answer;
    }
}